File: 733-flood-fill/733-flood-fill.py
class Solution:
    def floodFill(self, image: List[List[int]], sr: int, sc: int, newColor: int) -> List[List[int]]:
        R = len(image)
        C = len(image[0])
        color = image[sr][sc]
        if color == newColor: return image
        def ff(sr, sc):
            if image[sr][sc] == color:
                image[sr][sc] = newColor 
                if sr >=1: ff(sr-1, sc)
                if sr+1 < R: ff(sr+1, sc)
                if sc >=1: ff(sr, sc-1)
                if sc+1 < C: ff(sr, sc+1)
        ff(sr, sc)
        return image
            

File: 733-flood-fill/733-flood-fill.cpp
class Solution {
public:
    void dfs(int sr, int sc, vector<vector<int>> &image, int newColor, int sourceColor){
        int m = image.size();
        int n = image[0].size();
        if(image[sr][sc]==sourceColor){
            image[sr][sc] = newColor;
            if(sr > 0) dfs(sr - 1, sc, image, newColor, sourceColor);
            if(sr < m - 1) dfs(sr + 1, sc, image, newColor, sourceColor);
            if(sc > 0) dfs(sr, sc-1, image, newColor, sourceColor);
            if(sc < n - 1) dfs(sr, sc+1, image, newColor, sourceColor);
        }
        
    }
    
    
    
    vector<vector<int>> floodFill(vector<vector<int>>& image, int sr, int sc, int newColor) {
        int sourceColor = image[sr][sc];
        if(sourceColor == newColor) return image;
        
        dfs(sr, sc, image, newColor, sourceColor);
        return image;
        
        
    }
};